esmvaltool/diag_scripts/autoassess/land_surface_soilmoisture/soilmoisture.py

```diff
@@ -27,69 +35,117 @@ def get_provenance_record(caption, run):
+
+    Parameters
+    ----------
+    output_dir : string
+        The full path to the directory in which the CSV file will be written.
+    metrics : dictionary of metric,value pairs
+        The seasonal data to write.
+    config : dictionary
+        ESMValTool configuration object
+    """
+    os.makedirs(output_dir, exist_ok=True)
+
+    file_name = "metrics.csv"
+    file_path = os.path.join(output_dir, file_name)
+
+    with open(file_path, "w", newline="", encoding="utf-8") as csvfile:
+        csv_writer = csv.writer(csvfile)
+        for line in metrics.items():
+            csv_writer.writerow(line)
+
+    record_provenance(file_path, config)
+
+
+def volumetric_soil_moisture(model_file, constr_season):
+    """
+    Read moisture mass content and convert to volumetric soil moisture.
+
+    Parameters
+    ----------
+    model_file : string
+        Path to model file
+    constr_season : iris constraint
+        Constraint on season to load
+
+    Returns
+    -------
+    vol_sm1_run : cube
+        Volumetric soil moisture
+    """
+    # Constant: density of water
+    rhow = 1000.
+
+    # m01s08i223
+    # CMOR name: mrsos (soil moisture in top model layer kg/m2)
+    mrsos = iris.load_cube(
+        model_file,
+        "mass_content_of_water_in_soil_layer" & constr_season
+    )
+
+    # Set soil moisture to missing data where no soil (moisture=0)
+    np.ma.masked_where(mrsos.data == 0, mrsos.data, copy=False)
+
+    # first soil layer depth
+    dz1 = mrsos.coord('depth').bounds[0, 1] - \
+        mrsos.coord('depth').bounds[0, 0]
+
+    # Calculate the volumetric soil moisture in m3/m3
+    # volumetric soil moisture = volume of water / volume of soil layer
+    # = depth equivalent of water / thickness of soil layer
+    # = (soil moisture content (kg m-2) / water density (kg m-3) )  /
+    #      soil layer thickness (m)
+    # = mosrs / (rhow * dz1)
+    vol_sm1_run = mrsos / (rhow * dz1)
+    vol_sm1_run.units = "m3 m-3"
+    vol_sm1_run.long_name = "Top layer Soil Moisture"
+
+    return vol_sm1_run
+
+
+def land_sm_top(clim_file, model_file, model_dataset, config):
     """
     Calculate median absolute errors for soil mosture against CCI data.
 
     Parameters
     ----------
-    run: dict
-        dictionary containing model run metadata
-        (see auto_assess/model_run.py for description)
+    clim_file : string
+        Path to observation climatology file
+    model_file : string
+        Path to model file
+    model_dataset : string
+        Name of model dataset
+    config : dict
+        ESMValTool configuration object
 
     Returns
     -------
     metrics: dict
         a dictionary of metrics names and values
     """
-    supermean_data_dir = os.path.join(run['data_root'], run['runid'],
-                                      run['_area'] + '_supermeans')
+    # Input filenames for provenance
+    filenames = [item["filename"] for item in config["input_data"].values()]
```

Comment:
```suggestion
    filenames = [clim_file, model_file]
```
if I understand the code here correctly, the saved data is computed from the input data in `clim_file` and `model_file`, right? The point of [provenance](https://en.wikipedia.org/wiki/Provenance) is keeping track of what input data was used and how it was processed. In the currently implemented provenance, I see the E3SM-1-0 model listed as an ancestor to the data created for the ACCESS-CM2 model, but it looks like it is not actually used in the computation.

Reply:
Good point. i've followed the spirit of your suggestion to make the filenames more specific. Change (https://github.com/ESMValGroup/ESMValTool/pull/3197/commits/10cb722f91d09e42625ee69ad9ae76323f70af77) is a bit more extensive than your suggestion because `model_file` is actually a list of strings that needed to be flattened, and also now constructing the list at a higher level so as to pass to `write_metrics()` which had the same issue.